redash/query_runner/__init__.py

```diff
@@ -208,6 +217,17 @@ def _get_tables_stats(self, tables_dict):
                 res = self._run_query_internal("select count(*) as cnt from %s" % t)
                 tables_dict[t]["size"] = res[0]["cnt"]
 
+    @property
+    def supports_auto_limit(self):
+        return True
+
+    def apply_auto_limit(self, query_text):
+        query_list = sqlparse.split(query_text)
+        # we only check for last one in the list because it is the one that we show result
+        last_query = query_list[-1]
```

Comment:
I think this will not work correctly with the following:

```sql
select * from raw_events;
# comment
```

We should use the logic from #5058 once it's merged.

Reply:
It may become interesting when combining this feature and 5058. As far as I understand, we'll need to apply limit only to the last query (which we return to user), right?

Reply:
I see. I missed the case that user can type comment after query.

Reply:
But can databricks query runner execute such queries? I mean with a semicolon followed by a comment? Previously I was getting a syntax error in this case

Reply:
P.S. But looks that variant w/o semicolon may still be the case (last token is a comment): 
```sql
select * from raw_events
# comment
```

Reply:
Sure, I will update my code to use Levko's function once #5058 get merged into master.

Reply:
#5058 is merged.

Reply:
Sure, just update the code to use that function